pkg/planner/core/indexmerge_path.go

```diff
@@ -744,77 +554,44 @@ func generateIndexMerge4NormalIndex(ds *logicalop.DataSource, regularPathCount i
 		}
 	}
 
-	if !needConsiderIndexMerge {
-		return "IndexMerge is inapplicable or disabled. ", nil // IndexMerge is inapplicable
-	}
```

Comment:
Since this limitation only applies to non-MV indexes, and now the MV index path is also generated here, as said in the design doc, we modify this limitation a little and move it to the end of this function.

Reply:
It causes some extra unnecessary work now. This is also why there are new records in `pkg/planner/cardinality/testdata/cardinality_suite_out.json`.